clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -497,40 +501,18 @@ private synchronized void resetStateAndRejoin() {
             joinFuture.addListener(new RequestFutureListener<ByteBuffer>() {
                 @Override
                 public void onSuccess(ByteBuffer value) {
-                    // handle join completion in the callback so that the callback will be invoked
```

Comment:
Moved this into sync-group handler for readability.

Reply:
I assume you mean the JoinGroup response handler 🙂

Reply:
Well I should say part of that (the enabling of the heartbeat thread) is in JoinGroup response handler, while the rest (update metrics, etc) is in SyncGroup response handler.